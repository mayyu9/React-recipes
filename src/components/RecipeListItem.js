import React from 'react';
import PropTypes from 'prop-types';

const RecipeListItem = ({ recipe, onFavourited, onClick }) => (
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    key={recipe.id}
    // onClick={ props.onClick(recipe.id)}
    onClick={() => onClick(recipe.id)}
  >
    <span className="mr1">
      <input
        type="checkbox"
        onClick={e => {
          e.stopPropogation();
          onFavourited(recipe.id);
        }
        }
      />
    </span>
    <span>{recipe.name}</span>
    <span>{recipe.category}</span>
  </li>
);

RecipeListItem.propTypes = {
  recipe: PropTypes.array,
  onClick: PropTypes.func,
  onFavourited: PropTypes.func,
};

export default RecipeListItem;
