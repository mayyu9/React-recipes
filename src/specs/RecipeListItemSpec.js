import React from 'react';
import renderer from 'react-test-renderer';
import RecipeListItem from '../components/RecipeListItem';

const testRecipe = {
  id: 1,
  name: 'test Recipe',
  category: 'test category',
};

describe('<RecipeList />', () => {
  test('should not break if no recipe passed to RecipeListItem component', () => {
    const component = renderer.create(
      <RecipeListItem />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render when recipe passed to RecipeListItem component', () => {
    const component = renderer.create(
      <RecipeListItem recipe={testRecipe} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
