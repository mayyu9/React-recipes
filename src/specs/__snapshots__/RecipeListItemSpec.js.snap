// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> should not break if no recipe passed to RecipeListItem component 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    className="mr1"
  >
    <input
      onClick={[Function]}
      type="checkbox"
    />
  </span>
  <span />
  <span />
</li>
`;

exports[`<RecipeList /> should render when recipe passed to RecipeListItem component 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    className="mr1"
  >
    <input
      onClick={[Function]}
      type="checkbox"
    />
  </span>
  <span>
    test Recipe
  </span>
  <span>
    test category
  </span>
</li>
`;
